src/main/java/com/emc/pravega/perf/WriterWorker.java

```diff
@@ -55,33 +57,79 @@
-        CompletableFuture retFuture = null;
-        Random rand = new Random();
-
-        for (int i = 0; (i < eventsPerWorker) &&
-                (Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun); i++) {
-
-            // Construct event payload
-            String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
-            String payload = String.format("%-" + messageSize + "s", val);
-            String key;
-            if (isRandomKey) {
-                key = Integer.toString(workerID + rand.nextInt());
-            } else {
-                key = Integer.toString(workerID);
+        perf.benchmark();
+        return null;
+    }
+
+    private class eventsWriter implements performance {
+
+        public void benchmark() throws InterruptedException, ExecutionException {
+            CompletableFuture retFuture = null;
+            Random rand = new Random();
+
+            for (int i = 0; i < events; i++) {
+
+                // Construct event payload
+                String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
+                String payload = String.format("%-" + messageSize + "s", val);
+                String key;
+                if (isRandomKey) {
+                    key = Integer.toString(workerID + rand.nextInt());
+                } else {
+                    key = Integer.toString(workerID);
+                }
+
+                final Instant startTime = Instant.now();
+                retFuture = writeData(key, payload);
+                // event ingestion
+                retFuture = stats.recordTime(retFuture, startTime, payload.length());
             }
 
-            final Instant startTime = Instant.now();
-            retFuture = writeData(key, payload);
-            // event ingestion
-            retFuture = stats.recordTime(retFuture, startTime, payload.length());
+            flush();
+
+            //Wait for the last packet to get acked
+            retFuture.get();
         }
+    }
 
-        flush();
+    private class eventspersecWriter implements performance {
 
-        //Wait for the last packet to get acked
-        retFuture.get();
+        public void benchmark() throws InterruptedException, ExecutionException {
+            CompletableFuture retFuture = null;
+            Random rand = new Random();
 
-        return null;
+            while (Duration.between(StartTime, Instant.now()).getSeconds() < secondsToRun) {
+                final Instant loopStart = Instant.now();
+                long loopTime;
+
+                for (int i = 0; ((loopTime = Duration.between(loopStart, Instant.now()).toMillis()) < 1000) &&
+                        (i < events); i++) {
+
+                    // Construct event payload
+                    String val = System.currentTimeMillis() + ", " + workerID + ", " + (int) (Math.random() * 200);
```

Comment:
I don't think we need all this stuff for every write. We can have the `workerId` as routing key and a fixed payload of the desired size built outside of the loop (perhaps you could just append a sequenceId to the payload for verification purposes).

Reply:
Thanks Raul, But, I would like keep this because, every time i will write it the time as part of message so that , during reading i can get the full response time    (write-> read)